packages/webapp/src/components/Forms/GeneralRevenue/index.jsx

```diff
@@ -12,93 +12,151 @@
     mode: 'onChange',
     defaultValues: {
       [SALE_DATE]:
         (sale?.[SALE_DATE] && getLocalDateInYYYYDDMM(new Date(sale[SALE_DATE]))) ||
         persistedFormData?.[SALE_DATE] ||
         getLocalDateInYYYYDDMM(),
       [SALE_CUSTOMER]: sale?.[SALE_CUSTOMER] || persistedFormData?.[SALE_CUSTOMER] || '',
-      [VALUE]: !isNaN(sale?.[VALUE]) ? sale[VALUE] : null,
-      [NOTE]: sale?.[NOTE] || '',
+      [REVENUE_TYPE_ID]:
+        revenueTypeOptions?.find((option) => option.value === sale?.revenue_type_id) ||
+        revenueTypeOptions?.find(
+          (option) => option.value === persistedFormData?.[REVENUE_TYPE_ID],
+        ) ||
+        null,
+      [VALUE]: !isNaN(sale?.[VALUE])
+        ? sale[VALUE]
+        : !isNaN(persistedFormData?.[VALUE])
+        ? persistedFormData[VALUE]
+        : null,
+      [NOTE]: sale?.[NOTE] || persistedFormData?.[NOTE] || '',
+      ...customFormChildrenDefaultValues,
     },
   });
 
-  useHookFormPersist(getValues);
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isValid },
+    getValues,
+    watch,
+    control,
+    reset,
+    setValue,
+  } = reactHookFormFunctions;
+
+  //useHookFormPersist(getValues);
+  const selectedTypeOption = watch(REVENUE_TYPE_ID);
+
+  const readonly = view === 'read-only';
+  const disabledInput = readonly;
+  const disabledButton = !isValid && !readonly;
+
+  // TODO: LF-3680 fix isDirty for dynamic fields
+  // const disabledButton = (!isValid || !isDirty) && !readonly;
+
+  const customChildren = useCustomFormChildren(
+    reactHookFormFunctions,
+    sale,
+    currency,
+    disabledInput,
+    revenueTypes,
+    selectedTypeOption,
+  );
+
+  // Separating these into separate vs prop rendered nodes prevents form submission onClick for noSubmitButton
+  const noSubmitButton = (
+    <Button
+      color={'secondary'}
+      fullLength
+      disabled={disabledButton}
+      onClick={onClick}
+      type={'button'}
+    >
+      {' '}
+      {buttonText}
+    </Button>
+  );
+  const submitButton = (
+    <Button color={'primary'} fullLength disabled={disabledButton} type={'submit'}>
+      {' '}
+      {buttonText}
+    </Button>
+  );
 
   return (
     <Form
       onSubmit={handleSubmit(onSubmit)}
       buttonGroup={
         <>
-          {onClickDelete && (
-            <Button color={'secondary'} fullLength onClick={onClickDelete} type={'button'}>
-              {t('common:DELETE')}
-            </Button>
-          )}
-          <Button disabled={!isValid} fullLength type={'submit'}>
-            {t('common:SAVE')}
-          </Button>
+          {onClick && noSubmitButton}
+          {onSubmit && submitButton}
         </>
       }
     >
-      <PageTitle title={title} onGoBack={() => history.back()} style={{ marginBottom: '24px' }} />
+      <PageTitle
+        title={title}
+        onGoBack={() => {
+          handleGoBack(reset);
+        }}
+        style={{ marginBottom: '24px' }}
```

Comment:
If possible make this one CSS instead of inline styling too 🙏 

There are some more inline styles in this file as well that hopefully could be translated into CSS rules.

Reply:
Yeah -- it's another instance where the components were all made specifically to take in a style object instead of className={styles.X} which I am more used to. It would be nice to refactor everything to have better styling consistency -- any suggestions?

I know I am perpetuating bad practice but I definitely copied this from our other forms.

I could make a fancy selector like the other one but it seems like they were made to take in style this way.

Reply:
I'm cool with leaving as is for now, it is definitely a broader problem that we'll need to solve for!